TF-572: Update config via Camera Masking Screen

JordanBerni-UL opened the pull request:
- [x] Code Review
- [x] Developer Testing
- [x] QA Review
- [x] Test Plan updated
- [x] PO Review (Optional)
- [x] Update Release Notes

**Known Issues**

- Sliders only update on drag end
- Camera's sometimes flicker

Created https://ultrahaptics.atlassian.net/browse/TF-649 to look at BRS performance issues

JordanBerni-UL committed:
Merge branch 'develop' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
Merge branch 'feature/TF-571_TouchFree_Service_API_for_Tracking_Setting_Changes' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Update Tooling

JordanBerni-UL committed:
WebSettings: Reinstate web tooling

JordanBerni-UL committed:
Merge branch 'feature/TF-571_TouchFree_Service_API_for_Tracking_Setting_Changes' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Store masking values in state

JordanBerni-UL committed:
WebSettings: Set sliders value based on state

JordanBerni-UL committed:
Merge branch 'feature/TF-571_TouchFree_Service_API_for_Tracking_Setting_Changes' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
Merge branch 'develop' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Update tooling

JordanBerni-UL committed:
WebSettings: Remove single use functions

JordanBerni-UL committed:
WebSettings: Add 'Allow Analytics' toggle

JordanBerni-UL committed:
WebSettings: Update toggle css

JordanBerni-UL committed:
Merge branch 'feature/TF-571_TouchFree_Service_API_for_Tracking_Setting_Changes' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Update tooling

JordanBerni-UL committed:
Merge branch 'feature/TF-571_TouchFree_Service_API_for_Tracking_Setting_Changes' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
Merge branch 'feature/TF-571_TouchFree_Service_API_for_Tracking_Setting_Changes' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Update tooling

JordanBerni-UL committed:
Merge branch 'feature/TF-571_TouchFree_Service_API_for_Tracking_Setting_Changes' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Update tooling

JordanBerni-UL committed:
WebSettings: Revert tooling formatting

JordanBerni-UL committed:
WebSettings: Update tooling

JordanBerni-UL committed:
Merge branch 'feature/TF-571_TouchFree_Service_API_for_Tracking_Setting_Changes' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Mask camera on dragEnd

JordanBerni-UL committed:
Merge branch 'feature/TF-571_TouchFree_Service_API_for_Tracking_Setting_Changes' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Update tooling

JordanBerni-UL committed:
WebSettings: Toggle feeds based on allowImages

JordanBerni-UL committed:
WebSettings: Show sliders on sub camera feed

JordanBerni-UL committed:
WebSettings: Make labelled toggles work with TF

JordanBerni-UL committed:
WebSettings: Clear masking while dragging

JordanBerni-UL committed:
WebSettings: Change lens selection to be a toggle

JordanBerni-UL committed:
WebSettings: Only react to correct pointer events

JordanBerni-UL committed:
WebSettings: Display correct sub lens name

JordanBerni-UL committed:
WebSettings: Add global pointerMove listener

JordanBerni-UL committed:
Merge branch 'develop' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings:  Add listeners for more move types

JordanBerni-UL committed:
WebSettings: Remove global debug listeners

JordanBerni-UL committed:
Show both cameras in one canvas

NatGreenwood-UL commented on TF_Settings_Web/src/Components/Controls/ToggleSwitch.tsx:
```diff
@@ -4,32 +4,26 @@ import React from 'react';
-            }}
-        />
+        {/* eslint-disable-next-line @typescript-eslint/no-empty-function */}
+        <input type="checkbox" style={{ display: 'none' }} checked={value} onChange={() => {}} />
```
Can these ToggleSwitches no longer be changed? Seems like they wouldn't do anything on click now?

JordanBerni-UL replied:
The label around the ToggleSwitch takes care of what to do when the checkbox is clicked (TF doesn't trigger the onChange function and that would only apply to the actual switch element, not the surrounding label).

I will add this as a comment to make it more clear

NatGreenwood-UL commented on TF_Settings_Web/src/TouchFree/Configuration/ConfigurationManager.ts:
```diff
@@ -62,14 +62,14 @@ export class ConfigurationManager {
     public static RequestConfigFileChange(
         _interaction: Partial<InteractionConfig> | null,
```
All the changes to the TouchFree Tooling applied from here down in the listed changes need to be replicated across to the "main" version of the TouchFree tooling before we end up with really divergent versions.

NatGreenwood-UL replied:
Copying one atop the other they seem to have quite a few differences: 
![image](https://user-images.githubusercontent.com/56393090/183406677-fad5b983-fdb5-4e20-86f3-b1ef1d1bd113.png)

JordanBerni-UL committed:
WebSettings: Add comment to ToggleSwitch

JordanBerni-UL committed:
WebSettings: Update tooling

JordanBerni-UL committed:
WebSettings: Only render one feed

JordanBerni-UL committed:
WebSettings: Add hover state to lens toggle

JordanBerni-UL committed:
WebSettings: Decrease FPS for performance

JordanBerni-UL committed:
WebSettings: Add hover state to lens toggle

JordanBerni-UL committed:
WebSettings: Request tracking state until recieved

JordanBerni-UL committed:
Merge branch 'experiment/TF-572_BRS' into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Integrate Gabe's performance changes

JordanBerni-UL committed:
WebSettings: Only toggle Reverse Cam on mouse

Stop dragging on hands lost

JordanBerni-UL committed:
WebSettings: Fade in complete buttons

JordanBerni-UL committed:
WebSettings: Remove unused ref

JordanBerni-UL committed:
WebSettings: Fade in prompt using CSS

cartmouse reviewed (approved):
I've had a look over these latest changes and it all looks good to me code wise!

JordanBerni-UL committed:
WebSettings: Use CSS fade-in

JordanBerni-UL committed:
WebSettings: Reduce camera resolution

JordanBerni-UL committed:
WebSettings: Add button to test tracking

JordanBerni-UL committed:
WebSettings: Add more debug statements

JordanBerni-UL committed:
WebSettings: Log when we lose hands

GabrielHomewoodUL committed:
Service - Fix request that requires connection not resending message

GabrielHomewoodUL committed:
Merge branch 'feature/TF-572_Camera_Masking_Call_API' of https://github.com/ultraleap/TouchFree into feature/TF-572_Camera_Masking_Call_API

JordanBerni-UL committed:
WebSettings: Remove debug statements

JordanBerni-UL committed:
WebSettings: Centre calibrating text

JordanBerni-UL committed:
WebSettings: Performance improvements

JordanBerni-UL committed:
WebSettings: Add Tracking Service alerts

Disable the camera masking screen button if we can't connect to the tracking service

JordanBerni-UL committed:
WebSettings: Return from masking screen if socket closes

cartmouse commented:
Reviewed up to here, all looking good!
